File: avail-rust/docs/book/src/examples/transactions.rs
use avail_rust::{avail, error::ClientError, utils, Block, Options, SDK};

pub async fn run() -> Result<(), ClientError> {
	let sdk = SDK::new(SDK::local_endpoint()).await?;

	let account = SDK::alice()?;

	let dest = utils::account_id_from_str("5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty")?;
	let options = Some(Options::new().nonce(avail_rust::Nonce::BestBlockAndTxPool));
	let tx = sdk.tx.balances.transfer_keep_alive(dest, SDK::one_avail());
	let res = tx.execute_wait_for_inclusion(&account, options).await?;

	let block = Block::new(&sdk.online_client, res.block_hash).await?;

	// transaction_all_static, transaction_count, transaction_by_signer, transaction_by_signer_static
	// transaction_by_index, transaction_by_index_static, transaction_by_hash,
	// transaction_by_hash_static, transaction_by_app_id, transaction_by_app_id_static
	for tx in block.transactions.iter() {
		println!(
			"Tx Pallet name: {}, Tx Name: {}, Tx Hash: {:?}",
			tx.pallet_name()?,
			tx.variant_name()?,
			tx.hash()
		);

		for event in tx.events().await?.iter() {
			let Ok(event) = event else {
				return Ok(());
			};

			println!(
				"\tEvent Pallet name: {}, Event Name: {}",
				event.pallet_name(),
				event.variant_name()
			);
		}

		let balance_tx = tx.as_extrinsic::<avail::balances::calls::types::TransferKeepAlive>();
		if let Some(tx) = balance_tx.ok().flatten() {
			println!("Transfer dest: {:?}, value: {}", tx.dest, tx.value);
		}
	}

	Ok(())
}

File: avail-js/docs/book/src/examples/transactions.ts
import { SDK, Block, Events, CallData } from "./../../../../src/index"

export async function run() {
  const sdk = await SDK.New(SDK.localEndpoint())
  const api = sdk.api

  const account = SDK.alice()

  const dest = "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"
  const value = SDK.oneAvail()
  const tx = sdk.tx.balances.transferKeepAlive(dest, value)
  const res = (await tx.executeWaitForInclusion(account))._unsafeUnwrap()

  const block = await Block.New(api, res.blockHash)

  // transactionAll, transactionBySigner, transactionByIndex, transactionByHash, transactionByAppId
  for (const [index, tx] of block.transactionAll().entries()) {
    console.log(`Tx Pallet name: ${tx.method.section}, Tx Name: ${tx.method.method}, Tx Hash: ${tx.hash.toHex()}`)

    const eventRecords = await Events.fetchEvents(api, res.blockHash, index)
    for (const eventRecord of eventRecords) {
      console.log(`\tEvent Pallet name: ${eventRecord.event.section}, Event Name: ${eventRecord.event.method}`)
    }
    const balance_tx = CallData.Balances.TransferKeepAlive.decode(tx)
    if (balance_tx != null) {
      console.log(`Transfer dest: ${balance_tx.dest}, value: ${balance_tx.value}`)
    }
  }
}
